Content.Shared/Camera/CameraRecoilSystem.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Player;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Camera;
+
+[UsedImplicitly]
+public class CameraRecoilSystem : EntitySystem
+{
+    /// <summary>
+    ///     Maximum rate of magnitude restore towards 0 kick.
+    /// </summary>
+    private const float RestoreRateMax = 15f;
+
+    /// <summary>
+    ///     Minimum rate of magnitude restore towards 0 kick.
+    /// </summary>
+    private const float RestoreRateMin = 1f;
+
+    /// <summary>
+    ///     Time in seconds since the last kick that lerps RestoreRateMin and RestoreRateMax
+    /// </summary>
+    private const float RestoreRateRamp = 0.1f;
+
+    /// <summary>
+    ///     The maximum magnitude of the kick applied to the camera at any point.
+    /// </summary>
+    private const float KickMagnitudeMax = 2f;
+
+    private readonly ISawmill _log;
+
+    protected CameraRecoilSystem(IEntityManager entityManager)
+    : base(entityManager)
+    {
+        _log = Logger.GetSawmill($"ecs.systems.{nameof(CameraRecoilSystem)}");
+
+        SubscribeNetworkEvent<CameraKickEvent>(HandleCameraKick);
+    }
```

Comment:
Bit weird when all the other entitysystems just do it in initialize and not the constructor?

Reply:
Do you want me to move these 2 lines to the Initialize override?